src/main/java/com/powsybl/sld/server/SingleLineDiagramController.java

```diff
@@ -265,14 +277,21 @@ public ResponseEntity<Collection<String>> getAvailableSvgComponentLibraries() {
             @Parameter(description = "Network UUID") @PathVariable("networkUuid") UUID networkUuid,
             @Parameter(description = "Voltage levels ids") @RequestParam(name = "voltageLevelsIds", required = false) List<String> voltageLevelsIds,
             @Parameter(description = "Variant Id") @RequestParam(name = "variantId", required = false) String variantId,
-            @Parameter(description = "depth") @RequestParam(name = "depth", required = false) int depth) throws JsonProcessingException {
-        LOGGER.debug("getNetworkAreaDiagramSvg request received with parameter networkUuid = {}, voltageLevelsIds = {}, depth = {}", networkUuid, sanitizeParam(voltageLevelsIds.toString()), depth);
-        SvgAndMetadata svgAndMetadata = networkAreaDiagramService.generateNetworkAreaDiagramSvg(networkUuid, variantId, voltageLevelsIds, depth);
-        String svg = svgAndMetadata.getSvg();
-        Object additionalMetadata = svgAndMetadata.getAdditionalMetadata();
-        return OBJECT_MAPPER.writeValueAsString(
-                OBJECT_MAPPER.createObjectNode()
-                        .put(SVG_TAG, svg)
-                        .putPOJO(ADDITIONAL_METADATA, additionalMetadata));
+            @Parameter(description = "depth") @RequestParam(name = "depth", required = false) int depth) throws InterruptedException, ExecutionException {
+        CompletableFuture<String> futureNAD = CompletableFuture.supplyAsync(() -> {
+            try {
+                LOGGER.debug("getNetworkAreaDiagramSvg request received with parameter networkUuid = {}, voltageLevelsIds = {}, depth = {}", networkUuid, sanitizeParam(voltageLevelsIds.toString()), depth);
+                SvgAndMetadata svgAndMetadata = networkAreaDiagramService.generateNetworkAreaDiagramSvg(networkUuid, variantId, voltageLevelsIds, depth);
+                String svg = svgAndMetadata.getSvg();
+                Object additionalMetadata = svgAndMetadata.getAdditionalMetadata();
+                return OBJECT_MAPPER.writeValueAsString(
+                        OBJECT_MAPPER.createObjectNode()
+                                .put(SVG_TAG, svg)
+                                .putPOJO(ADDITIONAL_METADATA, additionalMetadata));
+            } catch (JsonProcessingException e) {
+                throw new PowsyblException("Failed to parse JSON response", e);
+            }
```

Comment:
can you add a test where you go in this catch to reach the coverage threshold

Reply:
looks like its unreachable code. I don't think we care about the coverage